### PR TITLE
Add support for feature flags in Prometheus

### PR DESCRIPTION
**Pull request by onprem**
Related to prometheus-operator/prometheus-operator#3878

### CODE REVIEW

File: jsonnet/kube-prometheus/components/prometheus.libsonnet [hunk 1]
```diff
@@ -276,6 +277,7 @@ function(params) {
         labels: p._config.commonLabels,
       },
       externalLabels: p._config.externalLabels,
+      enableFeatures: p._config.enableFeatures,
```
Comment (paulfantom) [inline jsonnet/kube-prometheus/components/prometheus.libsonnet:280]: We cannot expose all the features of prometheus-operator (yet) and I wonder if we should expose this one in an easier way. The flag is meant for the more experienced users as everything behind it [is considered experimental](https://prometheus.io/blog/2021/02/17/introducing-feature-flags/) and it still can be set by simply adding:

```jsonnet
prometheus+: {
  prometheus+: {
    spec+: {
      enableFeatures: []
    }
  }
}
```

@prometheus-operator/kube-prometheus-reviewers any ideas? :)

Comment (onprem) [reply jsonnet/kube-prometheus/components/prometheus.libsonnet:280]: I am not very fluent with jsonnet so it didn't even came to my mind that I can do what you suggested.
```
prometheus+: {
  prometheus+: {
    spec+: {
      enableFeatures: []
    }
  }
}
```

This serves my use-case, so I'll leave upto you for what to do with this PR. Thanks!

**Comment by onprem** [comment]
I am going ahead and closing this, let me know if there is any interest in opening this again.

**Comment by ArthurSens** [comment]
Personally, I'd vote for allowing the enable-feature param just for the sake of cleaner code 🙂. Since we already use those parameters for other things today, I do prefer:
```jsonnet
local kp =
  (import 'kube-prometheus/main.libsonnet') +
  {
  values+:: {
    prometheus+: {
      externalLabels: {
        cluster: 'cluster-1',
        enableFeature: ['feature-x', 'feature-y'],
      },
    },
  },
};

{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } 
```

Over:
```jsonnet
local kp =
  (import 'kube-prometheus/main.libsonnet') +
  {
  values+:: {
    prometheus+: {
      externalLabels: {
        cluster: 'cluster-1',
      },
    },
  },
  prometheus+: {
    prometheus+: {
      spec+: {
        enableFeatures: ['feature-x', 'feature-y']
      },
    },
  },
};

{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } 
```

**Comment by paulfantom** [comment]
Since there is interest in this, let's revive this feature.

**Comment by onprem** [comment]
I am happy to continue with the work here. I have reopened the PR and rebased it on main. LMK if anything else in needed :)

**Comment by paulfantom** [comment]
Thanks! :100:

**Pull request merged**